ArduCopter/commands.cpp

```diff
@@ -94,8 +93,39 @@ bool Copter::set_home(const Location& loc, bool lock)
     // log ahrs home and ekf origin dataflash
     Log_Write_Home_And_Origin();
 
-    // send new home location to GCS
+    // send new home and ekf origin to GCS
     gcs().send_home(loc);
+    gcs().send_ekf_origin(ekf_origin);
+
+    // return success
+    return true;
+}
+
+// sets ekf_origin if it has not been set.
+//  should only be used when there is no GPS to provide an absolute position
+//  returns true on success
+bool Copter::set_ekf_origin(const Location& loc)
+{
+    // only allow setting origin if motors are disarmed
```

Comment:
Why? The origin will be set if GPS fix is acquired while flying.

Reply:
Yes, I also wondered if we should allow setting the ekf origin while moving.  I can't think of a reason why we can't so I'll remove that restriction.